modules/mobile/app/models/mobile/v0/adapters/disability_rating.rb

```diff
@@ -29,7 +29,8 @@ def individual_ratings(response)
                 diagnostic_text: rating['diagnostic_text']
               )
             end
-          end.compact
+          end
+          individual_ratings.compact.sort_by { |rating| rating.effective_date || '1/1/1800' }.reverse!
```

Comment:
Should the date string be converted to an actual date? It's unclear to me whether this sort is truly working as expected or it happens to work because either 
- it's just comparing strings to strings and they happen to order correctly
- it's comparing datetimes to strings and they just happen to order correctly
- the tests are weak and it's not even using the default string value

The tests aren't explicit about testing the ordering, so it's difficult to know from reading the tests whether the ordering is correct or if the tests have just been rejiggered to pass.

Reply:
How do you suggest making the test better? I did consider converting them to datetimes explicitly but the string comparison seemed to be working correctly so I didn't bother

Reply:
I'll update to do the datetime explicit comparison because I do agree that it's safer

Reply:
For the tests, I'd probably do something like

```
describe 'sorting' do
  let(:ratings) { JSON.parse(response.body).dig('data', 'attributes', 'individualRatings') }
  it 'uses reverse order' do
    # you can either find some programmatic way of testing the order or just spell it out like:
    rating_dates = ratings.map { |rating| rating['effectiveDate'] }
    expect(rating_dates).to eq([date1, date2, date3])
  end
  
  it 'places nil effective dates last' do
```